### PR TITLE
Added a description for the "Classic" editor #2832

### LINKED ISSUE #2832: Add block description/help text to Classic Text
All blocks should have descriptions (see #2023). The Classic Text block is missing one.

### PR DESCRIPTION
**Pull request by DevinWalker**
## Description
Added a description for the "Classic Editor" block - Resolves #2832

## How Has This Been Tested?

- Manually

## Types of changes

- UX update

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

File: blocks/library/freeform/old-editor.js [hunk 1]
```diff
@@ -159,7 +165,16 @@ export default class OldEditor extends Component {
 	render() {
 		const { id } = this.props;
 
+		const inspectorControls = (
+			<InspectorControls key="inspector">
+				<BlockDescription>
+					<p>{ __( 'The Class Editor block allows you add content using a WYSIWYG editor.' ) }</p>
```
Comment (aduth) [inline blocks/library/freeform/old-editor.js]: Typo: "Class" -> "Classic"

File: blocks/library/freeform/old-editor.js [hunk 2]
```diff
@@ -159,7 +165,16 @@ export default class OldEditor extends Component {
 	render() {
 		const { id } = this.props;
 
+		const inspectorControls = (
```
Comment (aduth) [inline blocks/library/freeform/old-editor.js]: Minor: Personal preference, but I don't think the variable assignment adds clarity to justify separating the element from being inlined into the return array.

File: blocks/library/freeform/old-editor.js (hunk 1 above)
Comment (aduth) [reply blocks/library/freeform/old-editor.js]: Is "WYSIWYG" a widely understood acronym?

**Review by aduth — CHANGES_REQUESTED** [review]
Typo is my main concern here, though WYSIWYG could be technical jargon for many users and might warrant a rewording.

File: blocks/library/freeform/old-editor.js (hunk 1 above)
Comment (Soean) [reply blocks/library/freeform/old-editor.js]: The block title is `Classic Text`.
Maybe we should describe is like Core: `The Classic Text block gives you an editor that is similar to a word processor.`
https://github.com/WordPress/wordpress-develop/blob/4.8/src/wp-admin/edit-form-advanced.php#L366

**Comment by jasmussen** [comment]
Thank you for the PR! Aside from the changes aduth suggested, this looks good.

Perhaps an alternate description could be this:

> The Classic Text block brings back the classic editor WordPress used to come with. Used by old posts, or if you prefer to write in this.

Something like that? @karmatosed any suggestions?

File: blocks/library/freeform/old-editor.js (hunk 2 above)
Comment (DevinWalker) [reply blocks/library/freeform/old-editor.js]: @aduth I've removed the unnecessary var for clarity as suggested

**Comment by DevinWalker** [comment]
I've updated the description to be as @Soean suggested. My primary concern with @jasmussen is that it seems to discredit Gutenberg's ability with "brings back the classic editor WordPress used to come with" like this is a shim. Rather, I suggest we push forward and simply describe this block as an additional block without describing its past.

**Comment by jasmussen** [comment]
> I suggest we push forward and simply describe this block as an additional functionality without describing its past.

Sounds good to me. 

I'm still struggling with this description, though: 

> The Classic Text block gives you an editor that is similar to a word processor.

Gutenberg as a whole is an editor, and as much as Classic Text is crucial to Gutenberg, it's still "just" a block, and not an _editor_ itself. I suppose this was also a problem with my initial suggestion above. I'm also not sure how describing it as similar to a Word processer is descriptive. 

I'm sorry I don't really have a better suggestion, but let's break it down together and see if something comes up:

- It's a block
- You can insert it and write in it if you prefer, but it also is inserted by default for pre-Gutenberg posts
- This block is codewise very similar to the editable canvas that ships in the current WordPress editor, to the point where TinyMCE buttons (for inserting shortcodes and whatnot) can still be added to it by plugins. 
- It offers more of what TinyMCE comes with by default, than any other block in Gutenberg.
- It can contain multiple linebreaks inside a single block. 

We don't have to write more than a single paragraph, we don't have to include all the above, we just have to explain how it's different from a Paragraph.

Another attempt:

> The Classic Text block has more rich text features than the normal Paragraph block, and is backwards compatible with older WordPress posts.

**Comment by DevinWalker** [comment]
@jasmussen thank you for your thoroughly thought house response. Thinking about it your description here:

> The Classic Text block has more rich text features than the normal Paragraph block, and is backwards compatible with older WordPress posts.

1. This is time-based description because as Gutenberg becomes the norm the "is backwards compatible with older WordPress posts" will need to be removed or changed. 
2. Would mentioning the backwards compatibility add any value to the description? For instance, users would then know they can copy-and-paste old content in? If not, I'd say omit that part. 

Here's my stab at it:

> The Classic Text block has more rich text editing features than the Paragraph block such as the ability to add lists and block quotes within a single block.

**Comment by jasmussen** [comment]
Love your suggestion. Solid.

**Comment by DevinWalker** [comment]
Thanks! Rereading I'm not liking "block quotes within a single block" - sounds weird. I think omitting "block" would be fine since "blockquote" is actually an HTML element which more content writers don't know / care about. 

> The Classic Text block has more rich text editing features than the Paragraph block such as the ability to add lists and quotes within a single block.

**Comment by jasmussen** [comment]
Actually that's a good point. Also want to strike the balance between suggesting this block is "better" then the paragraph block, it's unlikely to be as flexible in the future when nesting plays a role.

**Comment by DevinWalker** [comment]
@jasmussen that's a good point. How do you suggest we do that?

> The Classic Text block has more rich text editing features than the Paragraph block such as the ability to add lists and quotes within a single block but lacks some of the flexibility.

**Comment by jasmussen** [comment]
Overall I think I may be overthinking it, it's just a description, better to get something out there, we can always adjust.

But the difference is rather technical, classic text is literally the old editor, wrapped and packaged to work in gutenberg. The benefit is just that, back compat, but it's at the cost of some user confusion, "which one should I use?"

Maybe a different, shorter approach for the time being. Perhaps just "The classic editor, in block form".

It doesn't address the user confusion, but perhaps that should be handled separately?

**Comment by DevinWalker** [comment]
Sounds good to me. I'll adjust it to "The classic editor, in block form."

**Comment by jasmussen** [comment]
👍 👍 I think that's probably good to now. Thanks so much for your PR, and patience. 

@aduth are your concerns addressed?

**Review by aduth — APPROVED** [review]
Code looks good 👍

**Pull request merged**